### PR TITLE
[dlrm] Unconditionally move inputs to the specified device

### PR DESCRIPTION
**Pull request by alanwaketan**
This commit moves the X.to one step above to better support devices other
than CUDA and CPU.

### CODE REVIEW

File: torchbenchmark/models/dlrm/__init__.py [hunk 1]
```diff
@@ -163,8 +163,8 @@ def __init__(self, device=None, jit=False):
 
         # Preparing data
         X, lS_o, lS_i, self.targets = next(iter(self.train_ld))
+        X = X.to(self.device)
         if self.device == "cuda":
-            X = X.to(self.device)
             lS_i = [S_i.to(self.device) for S_i in lS_i] if isinstance(lS_i, list) \
```
Comment (wconstab) [inline torchbenchmark/models/dlrm/__init__.py:168]: don't we also need to move S_i and S_o to lazy device?

maybe its better to just change the conditional to 
`if self.device != "cpu":`
?

Comment (alanwaketan) [reply torchbenchmark/models/dlrm/__init__.py:168]: Fixed.

**Review by wconstab — APPROVED** [review]

**Review by xuzhao9 — APPROVED** [review]

File: torchbenchmark/models/dlrm/__init__.py [hunk 2]
```diff
@@ -163,7 +163,7 @@ def __init__(self, device=None, jit=False):
 
         # Preparing data
         X, lS_o, lS_i, self.targets = next(iter(self.train_ld))
-        if self.device == "cuda":
+        if self.device != "cpu":
```
Comment (xuzhao9) [inline torchbenchmark/models/dlrm/__init__.py:166]: One question, can we just delete this condition? I guess if `self.device == "cpu"`, `X = X.to(self.device)` is just no-op?

Comment (alanwaketan) [reply torchbenchmark/models/dlrm/__init__.py:166]: Yea, seems even better.

**Pull request closed without merging**